src/Lecture3.hs

```diff
@@ -33,6 +33,8 @@ module Lecture3
     , appendDiff3
     , apply
     ) where
+import GHC.ST (STret)
```

Comment:
I see a GHC warning that this import is redundant ⚠️ 
I wonder, what was your use case for this? 🤔